functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/configuration/SkipAfterFailureCountTest.java

```diff
@@ -36,22 +36,26 @@ public void should_skip_all_remaining_tests_with_first_failure_when_ordering_ena
         project.applyAsCommits("Introduces error by changing return value");
 
         // when
-        final Map<Status, Long> resultCount = new LinkedHashMap<>();
-
-        project.build("container/impl-base")
+        final TestResults testResults = project.build("container/impl-base")
             .options()
                 .ignoreBuildFailure()
             .configure()
-            .run(resultCount);
-
+            .run();
 
         // then
-        final Long skippedCount = resultCount.get(SKIPPED);
-        assertThat(skippedCount).isGreaterThan(4);
+        assertThat(testResults.testsWithStatuses(SKIPPED))
+            .extracting(TestResult::getClassName, TestResult::getTestMethod)
+            .contains(
+                tuple("org.jboss.arquillian.container.impl.client.container.ContainerLifecycleControllerTestCase", "org.jboss.arquillian.container.impl.client.container.ContainerLifecycleControllerTestCase"),
```

Comment:
Why class names and test methods are the same? This looks odd when reading the test.

Reply:
This denotes it's skipped on class level.

Reply:
I know it because we discussed that. But does another reader will? I am not sure. We should make it more self-explanatory.

Reply:
may be add some comments to make it more self-explanatory wdyt?

Reply:
Are comments the best we can do here?

Reply:
Yeah I think so. I am thinking for following approach but it has drawbacks
If I removed verification of `TestMethod` name & keep verifying only `TestClass` with uniqueness, in future someone might change that Test with only one method, think of it's executing, then it's ignored due to `@Ignore`,it's not ignored due to `skipAfterFailureCount`, so for this case test should not pass, but it's passing only because of validation of only `className`

Reply:
I am not sure if there is a "future" in this fixed repo, as every test controls the source code of the target project precise enough.

Reply:
So what we could do here instead is to introduce custom assertion saying `assertThat(testResults.testsWithStatuses(SKIPPED)).hasSkippedClasses("...")` and within it we could have this detailed check with tuples and a comment if needed about why it is expected to be like that. So we stick to strict verification but from the use of assertions and by reading it, it will become clear what are the expectations